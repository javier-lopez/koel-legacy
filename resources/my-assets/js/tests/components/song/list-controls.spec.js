import each from 'jest-each'
import Component from '@/components/song/list-controls.vue'
import factory from '@/tests/factory'

describe('components/song/list-controls', () => {
  it('renders properly', () => {
    expect(shallow(Component)).toMatchSnapshot()
  })

  each([[[]], [[factory('song')]]]).test('allows shuffling all if less than 2 songs are selected', selectedSongs => {
    expect(
      shallow(Component, { propsData: { selectedSongs }})
        .click('.btn-shuffle-all')
        .hasEmitted('shuffleAll')
    ).toBe(true)
  })

  it('allows shuffling selected if more than 1 song are selected', () => {
    expect(shallow(Component, { propsData: {
      selectedSongs: factory('song', 3)
    }}).click('.btn-shuffle-selected').hasEmitted('shuffleSelected')).toBe(true)
  })

  it('displays the "Add To" menu', () => {
    expect(shallow(Component, { propsData: {
      selectedSongs: factory('song', 3)
    }}).has('.btn-add-to')).toBe(true)
  })

  it('allows clearing queue', () => {
    expect(shallow(Component, {
      data: () => ({
        fullConfig: {
          clearQueue: true
        }
      })
    }).click('.btn-clear-queue').hasEmitted('clearQueue')).toBe(true)
  })

  it('allows deleting current playlist', () => {
    expect(shallow(Component, {
      data: () => ({
        fullConfig: {
          deletePlaylist: true
        }
      })
    }).click('.btn-delete-playlist').hasEmitted('deletePlaylist')).toBe(true)
  })
})
