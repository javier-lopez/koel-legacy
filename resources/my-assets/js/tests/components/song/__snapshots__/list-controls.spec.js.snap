// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/song/list-controls renders properly 1`] = `
<div class="buttons song-list-controls"><button class="btn btn-orange btn-shuffle-all"><i class="fa fa-random"></i> All
  </button>
  <!---->
  <!---->
  <!---->
  <!---->
  <addtomenu-stub config="[object Object]" songs=""></addtomenu-stub>
</div>
`;
